### ISSUE TITLE
Build failure: sbclPackages.fset

### ISSUE THREAD

**Issue by lugeha** [NixOS/nixpkgs#318943]
### Steps To Reproduce

1. build sbclPackages.fset
2. 
### Build log

```
error: builder for '/nix/store/8xdcfi9saazn1l3jsw7f7shkvfd6yk97-sbcl-fset-20200925-git.drv' failed with exit code 1;
       last 25 log lines:
       > ; wrote /build/source/Code/defs-tmpGHU3ALSV.fasl
       > ; compilation finished in 0:00:00.004
       > WARNING:
       >    Lisp compilation had style-warnings while
       >    compiling #<CL-SOURCE-FILE "fset" "Code" "defs">
       > ; compiling file "/nix/store/5ksqigarfkqy9v276q6yj009d843d5nj-source/Code/port.lisp" (written 01 JAN 1970 12:00:01 AM):
       > ;
       > ; caught ERROR:
       > ;   READ error during COMPILE-FILE:
       > ;
       > ;     Lock on package SB-EXT violated when interning ONCE-ONLY while in package
       > ;     FSET.
       > ;   See also:
       > ;     The SBCL Manual, Node "Package Locks"
       > ;
       > ;     (in form starting at line: 364, column: 0, position: 11230)
       >
       > ; compilation aborted after 0:00:00.013
       > ;
       > ; compilation unit aborted
       > ;   caught 2 fatal ERROR conditions
       > ;   caught 1 ERROR condition
       > ;   caught 1 STYLE-WARNING condition
       > BUILD FAILED: COMPILE-FILE-ERROR while
       >               compiling #<CL-SOURCE-FILE "fset" "Code" "port">
```

### Additional context

This package is required for the nyxt browser.

Introduced by:
```

commit 1acef6e636e4111933dfd0213ee20ffaafe997c7
Author: Hraban Luyat <hraban@0brg.net>
Date:   Thu May 30 09:43:37 2024 -0400

    sbcl: 2.4.4 -> 2.4.5

commit ac82a513e55582291805d6f09d35b6d8b60637a1
Merge: d9493dcdc9fb de0fdc1c784e
Author: Domen Kožar <domen@dev.si>
Date:   Wed May 29 12:04:41 2024 +0200

    Merge pull request #313255 from Philipp-M/androidenv-fix-ndk-toolchains
    
    androidenv: fix NDK toolchain linking issues
```

Package builds using commit previous to updating sbcl package.

### Notify maintainers

Hraban Luyat <hraban@0brg.net>

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.2-zen1, NixOS, 24.11 (Vicuña), 24.11.20240531.5f6ccfa`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.22.1`
 - nixpkgs: `/nix/store/7ms48fnylg1vjpy8av4sc52ybjlw9r8m-source`
```

**Comment by lugeha** [comment]
Fixed upstream

https://github.com/slburson/fset/commit/67eca709c2a6275692381fbaf018876b2ab11244

**Comment by FliegendeWurst** [comment]
Fixed in #319688
